### PR TITLE
Add Semi-SL for cls Benchmark Test

### PR DESCRIPTION
**Pull request by harimkang**
### Summary

- Add TVModels into Singe-label Benchmark Test
- Add Semi-SL Benchmark Test

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by chuneuny-emily — APPROVED** [review]

**Review by sungchul1 — APPROVED** [review]

File: tests/perf/test_classification.py [hunk 1]
```diff
@@ -258,3 +261,121 @@ def test_perf(
...
+        Benchmark.Model(task="classification/multi_class_cls", name="tv_efficientnet_v2_l_semisl", category="other"),
+        Benchmark.Model(task="classification/multi_class_cls", name="tv_mobilenet_v3_small_semisl", category="other"),
+    ]
+
+    DATASET_TEST_CASES = (
+        [
+            Benchmark.Dataset(
+                name=f"cifar10@{num_label}_{idx}",
+                path=Path(f"multiclass_classification/semi-sl/cifar10@{num_label}_{idx}/supervised"),
+                group="cifar10",
```
Comment (eunwoosh) [inline tests/perf/test_classification.py:285]: Group should be one of "small", "medium", "large". Could you update it?

File: tests/perf/test_classification.py [hunk 2]
```diff
@@ -258,3 +261,121 @@ def test_perf(
...
+                },
+            )
+            for idx in (0, 1, 2)
+            for num_label in (4, 10, 25)
+        ]
+        + [
+            Benchmark.Dataset(
+                name=f"svhn@{num_label}_{idx}",
+                path=Path(f"multiclass_classification/semi-sl/svhn@{num_label}_{idx}/supervised"),
+                group="svhn",
```
Comment (eunwoosh) [inline tests/perf/test_classification.py:304]: Same as above.

File: tests/perf/test_classification.py [hunk 3]
```diff
@@ -258,3 +261,121 @@ def test_perf(
...
+                },
+            )
+            for idx in (0, 1, 2)
+            for num_label in (4, 10, 25)
+        ]
+        + [
+            Benchmark.Dataset(
+                name=f"fmnist@{num_label}_{idx}",
+                path=Path(f"multiclass_classification/semi-sl/fmnist@{num_label}_{idx}/supervised"),
+                group="fmnist",
```
Comment (eunwoosh) [inline tests/perf/test_classification.py:323]: Same as above.

File: tests/perf/test_classification.py [hunk 4]
```diff
@@ -258,3 +261,121 @@ def test_perf(
...
+                extra_overrides={
+                    "train": {
+                        "data.config.train_subset.subset_name": "train_data",
+                        "data.config.val_subset.subset_name": "val_data",
+                        "data.config.test_subset.subset_name": "val_data",
+                        "deterministic": "True",
+                    },
+                },
+            )
+            for idx in (0, 1, 2)
```
Comment (eunwoosh) [inline tests/perf/test_classification.py:297]: If current configuration is used, 9 cases of every each dataset is tested with every models, which can make performence test quite longer. I think it's better to divide dataset in to 3 types("small", "medium", "large") and each type is corresponding to single case. How do you think about it?

File: tests/perf/test_classification.py [hunk 5]
```diff
@@ -258,3 +261,121 @@ def test_perf(
             criteria=self.BENCHMARK_CRITERIA,
             resume_from=fxt_resume_from,
         )
+
+
+class TestPerfSemiSLMultiClass(PerfTestBase):
+    """Benchmark single-label classification for Semi-SL task."""
+
+    MODEL_TEST_CASES = [  # noqa: RUF012
+        Benchmark.Model(task="classification/multi_class_cls", name="efficientnet_b0_semisl", category="speed"),
```
Comment (eunwoosh) [inline tests/perf/test_classification.py:270]: It seems that two models are set to `speed` and no model for `balance`. Is this your intention?

**Review by eunwoosh — COMMENTED** [review]
Thanks for your work. I left some comments. Could you check them?

File: tests/perf/test_classification.py (hunk 4 above)
Comment (harimkang) [reply tests/perf/test_classification.py:297]: I discussed this with Emily and we decided to go this way.

File: tests/perf/test_classification.py (hunk 1 above)
Comment (harimkang) [reply tests/perf/test_classification.py:285]: As far as I know, this is just a str type to collect the results in Summary, so I think it's okay to leave it as it is, but what do you think? Will this cause errors elsewhere?

The dataset used here doesn't have a clear distinction between small, medium, and large.

File: tests/perf/test_classification.py (hunk 5 above)
Comment (harimkang) [reply tests/perf/test_classification.py:270]: I'll update this later when see the results.

File: tests/perf/test_classification.py (hunk 1 above)
Comment (eunwoosh) [reply tests/perf/test_classification.py:285]: Currently, only "small", "medium", "large" and "all" can be passed to `data_group` arguments. Please check "tests/conftest.py". So, only if "all" is passed to the argument, these datasets won't be skipped. If you think it's ok, then I have objection to leave it as it is.

**Pull request merged**